doc/Onboarding/Markup/Overview.md

```diff
@@ -0,0 +1,222 @@
+
+```cs
+new TextBlock().Text(() => vm.CounterValue)
+```
+
+At this point you might be wondering what the `vm` is. C# Markup provides a strongly typed API for setting the `DataContext` of a control, as shown in the following example that sets the `DataContext` to the ViewModel. The `vm` is a placeholder reference for the ViewModel type that you provide to the `DataContext` extension.
+
+```cs
+.DataContext(
+    new MainViewModel(), 
+    (page, vm) => page
+        .Content(
+            new TextBlock().Text(() => vm.CounterValue)
+        )
+);
+```
+
+We refer to `vm` as a placeholder because at this point it is not actually a reference to the ViewModel. It is simply a placeholder that allows us to provide a strongly typed API for data binding. Rather than invoking the code `vm.CounterValue` when the `TextBlock` is created, the expression tree, `() => vm.CounterValue` is used to create a binding expression that will be evaluated when the `DataContext` for the `TextBlock` is set.
+
+In the above example, an instance of `MainViewModel` is provided to the `DataContext` extension method. If an instance of the ViewModel isn't available at the time the `DataContext` is set, you can provide the type of the ViewModel instead, as shown in the following example.
+
+```cs
+.DataContext<MainViewModel>(
+    (page, vm) => page
+        .Content(
+            new TextBlock().Text(() => vm.CounterValue)
+        )
+);
+```
+
+Currently the data binding is directly on the `CounterValue` property meaning that the text shown in the `TextBlock` will just be the value of the `CounterValue` property. However, we want to display the text "Counter: {value}" where {value} is the value of the `CounterValue` property. To do this we can provide a delegate that will be invoked each time the value of the `CounterValue` property changes, as shown in the following example.
+
+```cs
+new TextBlock().Text(() => vm.CounterValue, count => $"Counter: {count}")
+```
+
+In some case you need more control over the binding, such as when you need to change the binding mode or provide a converter. In these cases you can use the `Bind` extension method, as shown in the following example that two-way data binds the `StepSize` property to the `Text` property of a `TextBox`. 
+
+```cs
+new TextBox().Text(x => x.Bind(() => vm.StepSize).TwoWay()),
+```
+
+### Resources
+
+C# Markup provides a strongly typed API to get, create and add both static and theme resources. For example, to set the `Background` to theme resource get the `ApplicationPageBackgroundThemeBrush` from the default WinUI Fluent theme:
+
+```cs
+this.Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"));
+```
+
+There are also predefined resources for theme brushes, as shown in the following example:
+
+```cs
+this.Background(Theme.Brushes.Background.Default);
+```
```

Comment:
These are NOT the same and we should not be indicating for any reason that they are. Uno.Themes.WinUI.Markup is based on the Material theme resource keys and is not usable with the default Fluent resources.

https://github.com/unoplatform/Uno.Themes/blob/0b2a7a98bca04d84285747390d046709347f335d/src/library/Uno.Themes.WinUI.Markup/Theme.Brushes.cs#L705-L706